src/main/java/com/googleinterns/zoomtube/servlets/DiscussionServlet.java

```diff
@@ -62,34 +64,29 @@ public void init() throws ServletException {
   @Override
   public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
     long lectureId = Long.parseLong(request.getParameter(PARAM_LECTURE));
-    Key lecture = KeyFactory.createKey(/* kind= */ "Lecture", lectureId);
-
+    Key lecture = KeyFactory.createKey(LectureUtil.KIND, lectureId);
     String parentIdString = request.getParameter(PARAM_PARENT);
     Key parent = null;
     if (parentIdString != null) {
       long parentId = Long.parseLong(parentIdString);
-      parent = KeyFactory.createKey(Comment.ENTITY_KIND, parentId);
+      parent = KeyFactory.createKey(CommentUtil.KIND, parentId);
     }
-
     User author = userService.getCurrentUser();
     if (author == null) {
       response.sendError(HttpServletResponse.SC_FORBIDDEN, "You are not logged in.");
       return;
     }
-
     String content = CharStreams.toString(request.getReader());
 
-    Entity commentEntity = new Entity(Comment.ENTITY_KIND);
-    commentEntity.setProperty(Comment.PROP_LECTURE, lecture);
-    commentEntity.setProperty(Comment.PROP_PARENT, parent);
-    // TODO: Add support for timestamped comments
-    commentEntity.setProperty(Comment.PROP_TIMESTAMP, new Date(0));
-    commentEntity.setProperty(Comment.PROP_AUTHOR, author);
-    commentEntity.setProperty(Comment.PROP_CONTENT, content);
-    // We set creation time to the time of the post request (now).
-    commentEntity.setProperty(Comment.PROP_CREATED, new Date(Clock.systemUTC().millis()));
-
-    datastore.put(commentEntity);
+    final Entity comment;
+    if (parent == null) {
+      comment = CommentUtil.createEntityNoParent(
+          lecture, new Date(0), author, content, new Date(Clock.systemUTC().millis()));
+    } else {
+      comment = CommentUtil.createEntityWithParent(
+          lecture, parent, new Date(0), author, content, new Date(Clock.systemUTC().millis()));
+    }
+    datastore.put(comment);
```

Comment:
For simplicity I would do the following:

if (parent == null) {
  datastore.put(CommentUtil.createEntityWithNoParent(...);
  return;
}
datastore.put(CommentUtil.createEntity(...);

Reply:
Cleaned this up.  I had to duplicate one line of code but got rid of the else

Reply:
I think this response.setStatus was added. Because of that I would just use a if/else statement so this line does not need to be duplicated.